docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -112,6 +112,9 @@ for the default implementation would not preclude the use of alternative control
 - To support auto remediation triggered by MachineHealthCheck objects:
   E.g. a MachineHealthCheck marks a machine for remediation. Control Plane Provider reconciles by removing the machine and replaces it with a new one
   **if and only if** the operation is not potentially destructive for the cluster (e.g. the operation could cause a permanent quorum loss).
+- To enable scale in (delete-and-create) during control plane upgrade and default current scale out (create-swap-and-delete) rollout strategy.
+- To support easy addition of new upgrade strategies.
```

Comment:
I suggest to be generic and add "- To support different upgrade strategies." right after "- To enable declarative orchestrated replacement of control plane machines, such as to roll out an OS-level CVE fix".

Otherwise, given that we are adding scale in as a goal, for completeness, we should document scale out as well, but TBH this seems like moving implementation details into the goal section